Polished the latest update blog (#3390)
transloadit/uppy: website/src/_posts/2021-12-2.1-2.3.md

published: true
---

> Last Christmas, we gave you [Uppy 1.24](https://uppy.io/blog/2020/12/1.24/), but this very next year, we’ll take it away (since it’s outdated by now) and give you a brand-new Uppy 2.3 🎁!

Reviewer: This line doesn't pass the linter validation.

Author: Why could that be?

I'm not exactly sure what linter validation is, but I don't think I made erroneous edits here.

Reviewer: Our _very_ picky linter thinks we shouldn't use `very` in our docs – I suppose our tooling doesn't know about Mariah Carey 🙄😅
This fix is to add a `<!--retext-simplify ignore very-->` comment before the sentence to silence the error.

## `@uppy/unsplash` is now production ready

The Unsplash plugin has received an update, providing attribution to the image authors and adding some polish to make it look better. Being fully brand compliant with Unsplash means it’s now production ready. You can demo the plugin on the [website](https://uppy.io/examples/dashboard) or read the [docs](https://uppy.io/docs/unsplash/).

## New streaming interface for Companion

[Companion][companion] handles the server-to-server communication between your server and file storage providers such as Google Drive, Dropbox, Instagram, and the like. This bypasses the client (so a 5 GB video isn’t eating into your users’ data plans) and files are directly uploaded to the final destination. Companion is made to be effortlessly integrate-able with Uppy.

> It works by downloading the requested file from a remote provider on the server to then start uploading it to the destination. This does the job, but it’s not ideal.

Author: This sentence still confused me a bit. Shouldn't this say: 'It use to work by..."? Or is this not a description of the functionality before the update?

It's also not completely clear to me what 'it' refers to. I suppose 'Companion'?

Reviewer 2: Yes, good point about “it used to”.

> It works by downloading the requested file from a remote provider on the server to then start uploading it to the destination. This does the job, but it’s not ideal.

Reviewer 2: ```suggestion
It used to work by downloading the requested file from a remote provider on the server to then start uploading it to the destination. This did the job, but it’s wasn’t ideal.
```

You can enable this by setting `streamingUpload` in the Companion [options](https://uppy.io/docs/companion/#Options).

**NOTE:** Streaming is turned off by default. Do not turn it on yet if you’re using TUS for uploads, due to [this tus-js-client bug](https://github.com/tus/tus-js-client/issues/275).

Interested in using Companion? You can use Transloadit-hosted Companion on every [Plan](https://transloadit.com/pricing/).

## Status Bar error state improvements

The Status Bar plugin would get confused about upload errors in Uppy, and we’ve addressed this:

* The error state is now shown in case of only _one_ failed file instead of _all_ failed files. The plugin used to get stuck in the uploading state when a single file failed.
* Added “X of X files uploaded” below “Upload failed” for extra context.
* Improved the error details button styling in the Status Bar and the file info card.
* Status Bar state is set to complete if the user manually removes the failed files.

![status bar improvements screenshot](/images/blog/2.1-2.3/status-bar-improvements.jpg)

## Internal housekeeping

> ### Yarn v3 instead of NPM

Reviewer: I don't think this is correct, npm themselves always use it all lowercase (e.g. https://docs.npmjs.com/about-npm)

Author: Ah alright. This was an edit for consistency. The second paragraph of this blog contains NPM written in upper case, so I just followed that example here.
